src/components/SignInButton.tsx

```diff
@@ -118,13 +128,6 @@ export const SignInButton = (props: SignInButtonProps) => {
 
   const shouldRedirectTo = (page: string): boolean => page !== '/' && page !== '/home';
 
-  const attemptSignInCheckToSAndRedirect = async (redirectTo:string, shouldRedirect: boolean) => {
-    await checkToSAndRedirect(shouldRedirect ? redirectTo : null);
-    Metrics.identify(Storage.getAnonymousId());
-    Metrics.syncProfile();
-    Metrics.captureEvent(eventList.userSignIn);
```

Comment:
I moved these to `checkToSAndRedirect` in the place where we know the user is authenticated and accepted ToS.